main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -541,18 +546,20 @@ object Lowering {
     case Expression.NewChannel(exp, tpe, pur, eff, loc) =>
       val e = visitExp(exp)
       val t = visitType(tpe)
-      Expression.NewChannel(e, t, pur, eff, loc)
+      val newChannel = Expression.Def(Defs.ChannelNew, Type.mkImpureArrow(e.tpe, t, loc), loc)
+      Expression.Apply(newChannel, e :: Nil, t, pur, eff, loc)
 
     case Expression.GetChannel(exp, tpe, pur, eff, loc) =>
       val e = visitExp(exp)
       val t = visitType(tpe)
-      Expression.GetChannel(e, t, pur, eff, loc)
+      val getChannel = Expression.Def(Defs.ChannelGet, Type.mkImpureArrow(e.tpe, t, loc), loc)
+      Expression.Apply(getChannel, e :: Nil, t, pur, eff, loc)
 
-    case Expression.PutChannel(exp1, exp2, tpe, pur, eff, loc) =>
+    case Expression.PutChannel(exp1, exp2, _, pur, eff, loc) =>
```

Comment:
This `tpe` is always `Unit` so you could remove it from the case class arguments and put it as a constant instead